.github/workflows/create-release.yml

```
@@ -31,6 +31,22 @@ jobs:
           python -m pip install bump2version
           echo -n "::set-output name=next_tag::"
           bump2version --list ${{ github.event.inputs.versionName }} | grep new_version | sed -r s,"^.*=",,
+      - name: Get main branch SHA
+        id: base-sha
+        run: echo "sha=$(git rev-parse origin/main)" >> $GITHUB_OUTPUT
+      - name: Get Python client file changes
+        id: changed-py
+        uses: tj-actions/changed-files@v44
+        with:
+            base_sha: "${{ steps.base-sha.outputs.sha }}"
+            files: |
+              src/client/delphi_epidata.py
+              src/client/packaging/pypi/**
+      - name: Bump Python versions if client files changed
+        if: steps.changed-py.outputs.any_changed == 'true'
+        run: |
+          cd src/client/packaging/pypi
+          bump2version --allow-dirty --new-version ${{ steps.version.outputs.next_tag }} ignore_part
```

please make a comment that notes this is using `${{ steps.version.outputs.next_tag }}`, the new literal version number output from the `bump2version` step above, not the workflow input `${{ github.event.inputs.versionName }}` which can be relative, thus ensuring the version numbers stay in sync.

it looks like `ignore_part` is just a placeholder for the otherwise required "part" field that is being overridden by `--new-version`...  can we simplify this to something closer in format to our other uses of `bump2version`?
```suggestion
          bump2version --allow-dirty ${{ steps.version.outputs.next_tag }}
```

also, is `--allow-dirty` necessary here?

---

> please make a comment that notes this is using `${{ steps.version.outputs.next_tag }}`, the new literal version number output from the `bump2version` step above, not the workflow input` ${{ github.event.inputs.versionName }}` which can be relative, thus ensuring the version numbers stay in sync.

Will do!

> it looks like `ignore_part` is just a placeholder for the otherwise required "part" field that is being overridden by `--new-version`... can we simplify this to something closer in format to our other uses of `bump2version`?

Unfortunately no, this is a quirk of the tool: https://github.com/c4urself/bump2version/issues/22

> also, is `--allow-dirty` necessary here?

It is, because we use `bump2version` twice: first to bump the versions for other than the Python client (which makes the working directory "dirty"), then the Python client as well.

---

> > it looks like `ignore_part` is just a placeholder for the otherwise required "part" field that is being overridden by `--new-version`... can we simplify this to something closer in format to our other uses of `bump2version`?
> 
> Unfortunately no, this is a quirk of the tool: https://github.com/c4urself/bump2version/issues/22
> 

wow, that is disappointing on multiple levels: bump2version is arguably broken, and this line has (and the 9 other places in the org that its been copied to have too) been lying to us about acceptable inputs: https://github.com/cmu-delphi/delphi-epidata/blob/69835d1d7795eaf9a710d9f4903fef22a07e8fdf/.github/workflows/create-release.yml#L5-L7

---

FWIW, bump2version says it's no longer maintained and that [this](https://github.com/callowayproject/bump-my-version) is the maintained fork.

---

[Tried it](https://github.com/cmu-delphi/delphi-epidata/pull/1473) and eh, it still has issues.